internal/controller/apiextensions/claim/reconciler.go

```diff
@@ -505,8 +521,20 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 		return reconcile.Result{Requeue: true}, errors.Wrap(r.client.Status().Update(ctx, cm), errUpdateClaimStatus)
 	}
 
-	err := r.client.Apply(ctx, cp, resource.AllowUpdateIf(func(old, obj runtime.Object) bool { return !cmp.Equal(old, obj) }))
+	var err error
+	if meta.WasCreated(cp) {
+		err = r.client.Apply(ctx, cp, resource.AllowUpdateIf(func(old, obj runtime.Object) bool { return !cmp.Equal(old, obj) }))
+	} else {
+		// if composite did not exist at the beginning of the loop, we want to create it
+		// so that we can check if the composite name is not already taken
+		err = r.client.Create(ctx, cp)
+	}
 	switch {
+	case kerrors.IsAlreadyExists(err):
```

Comment:
Just nits: 

- this could be thrown only from the Create above, so bugging me to have here for both.
- Would it be possible to convert the  above `else` to an [early return](https://github.com/crossplane/crossplane/tree/master/contributing#return-early) somehow 🤔